pkg/clusterconfig/clusterconfig.go

```diff
@@ -92,13 +102,59 @@ func Get() (*Config, error) {
 
 	cfg := &Config{}
 
-	cm, err := client.ConfigMaps(configNamespace).Get(configName, metaapi.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			logrus.Warnf("config-map %s/%s not found", configNamespace, configName)
-			return cfg, nil
+	cm, cmerr := client.ConfigMaps(configNamespace).Get(configName, metaapi.GetOptions{})
+
+	if cmerr != nil && errors.IsNotFound(cmerr) {
+		logrus.Warnf("config-map %s/%s not found, creating", configNamespace, configName)
+		cm = &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: corev1.SchemeGroupVersion.String(),
+				Kind:       "ConfigMap",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configName,
+				Namespace: configNamespace,
+			},
+			Data: make(map[string]string),
+		}
+	}
```

Comment:
refactor this and the logic below that attempts to create/update it, into a single helper that is CreateOrUpdateConfigMap (also why aren't these configmap keys simply part of the S3 storage struct on the registry resource?)  It's just the region value right?

Reply:
global config no longer needed, most of this is removed in the next push